### DISCUSSION TITLE
PageContext in component

### DISCUSSION THREAD

**Discussion by Nevinia210** [microsoft-search/pnp-modern-search#3046] [Q&A] [answered]
Hello,
i started working on custom layout for search results and i have one question.
how do you pass PageContext from this [example](https://microsoft-search.github.io/pnp-modern-search/extensibility/#accessing-the-sharepoint-framework-context-and-services-in-a-library-component) to component rendered in IComponentDefinition?
I have component written according to this [documentation](https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_web_component/#create-the-component-logic-and-sub-components), but i dont know how to pass/create pagecontext there to initialize spfi

**Comment by patrikhellgren** [comment, marked as answer]
@Nevinia210 You can do it in the following way in the web component class:

``` TypeScript
  public async connectedCallback() {
    let props = this.resolveAttributes();

    let serviceScope: ServiceScope = this._serviceScope; // Default is the root shared service scope regardless the current Web Part 

    if (props.instanceId) {

        // Get the service scope and keys associated to the current Web Part displaying the component
        serviceScope = this._webPartServiceScopes.get(props.instanceId) ? this._webPartServiceScopes.get(props.instanceId) : serviceScope;
    }

    const pageContext = serviceScope.consume<PageContext>(PageContext.serviceKey);

    const exampleComponent: React.ReactElement<IExampleComponentProps> = React.createElement(
      ExampleComponent,
      {
        pageContext,
        ...props
      }
    );

    ReactDOM.render(exampleComponent, this);
```

And then in ExampleComponent do like this to initialize a web object to a search result item´s site url using PnP JS:

``` TypeScript
const web = Web(item["SPSiteUrl"]).using(SPFx({pageContext: props.pageContext})).using(Caching());
```

**Reply by Nevinia210** [reply]
@patrikhellgren works like a charm :) thank you
